cudax/include/cuda/experimental/__async/basic_sender.cuh

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright (c) 2024 NVIDIA Corporation
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cuda/std/__utility/move.h>
+
+#include "completion_signatures.cuh"
+#include "config.cuh"
+#include "cpos.cuh"
+#include "utility.cuh"
+
+// This must be the last #include
+#include "prologue.cuh"
+
+namespace cuda::experimental::__async
+{
+template <class Data, class Rcvr>
+struct state
```

Comment:
Right now everything here is non-ugly. I am fine with leaving it as is, but we will need to fix that in the near future

Reply:
everything is within the ugly `__async` namespace. is that enough?

Reply:
Nope, because of evil user macros

Reply:
everything is terrible